language/move-lang/src/parser/ast.rs

```diff
@@ -174,17 +174,42 @@ pub struct SpecBlock_ {
     pub members: Vec<SpecBlockMember>,
 }
 
+pub type SpecBlock = Spanned<SpecBlock_>;
+
 #[derive(Debug, PartialEq)]
 pub enum SpecBlockTarget_ {
     Code,
     Module,
     Function(FunctionName),
     Structure(StructName),
+    Schema(Name, Vec<(Name, Kind)>),
 }
 
 pub type SpecBlockTarget = Spanned<SpecBlockTarget_>;
 
-pub type SpecBlock = Spanned<SpecBlock_>;
+#[derive(Debug, PartialEq)]
+pub struct PragmaProperty_ {
+    pub name: Name,
+    pub value: Option<Value>,
+}
+
+pub type PragmaProperty = Spanned<PragmaProperty_>;
+
+#[derive(Debug, PartialEq)]
+pub struct FunctionPattern_ {
+    pub name_pattern: Vec<NamePatternFragment>,
+    pub type_arguments: Option<Vec<Type>>,
+}
+
+pub type FunctionPattern = Spanned<FunctionPattern_>;
+
+#[derive(Debug, PartialEq)]
+pub enum NamePatternFragment_ {
+    Wildcard,
+    NamePart(Name),
+}
+
+pub type NamePatternFragment = Spanned<NamePatternFragment_>;
```

Comment:
Similar issue as above. Maybe  `Spec...Pattern..._`? I think the term pattern by itself is a bit weird if its not associated with expression pattern matching

Reply:
Done as well.